docs/setup/k8s/running-in-production-k8s.md

```diff
@@ -0,0 +1,122 @@
+These guidelines refer to more advanced use cases and they assume the surrounding work (read: devops) can be done by the targeted user.
+We know these things can get a bit tricky. If you want access to a fully-managed, air-gap-friendly, batteries-included, Nuclio deployment, packaged with **Lots** of other goodies - do check out the enterprise grade [Iguazio Data science PaaS](https://www.iguazio.com/platform/)! 
+
+That being said, here are a few guidelines to get you on your way:
+
+- Most definitely [version freeze](#version-freezing). Also, use `*.image.pullPolicy` to `Never` or `IfNotPresent` to make sure k8s won't try to access the web to fetch images at any point in time.
+- Set `Values.offline=true` in the helm values, to put nuclio in "offline" mode. Set `dashboard.baseImagePullPolicy=Never`.
+- Needless to say, in this scenario, you will configure nuclio with `registry.pushPullUrl` which is reachable from your system.
+- The processor and onbuild images will also have to be available to the dashboard in your environment, as they must be available for the building process - (by `docker build`, or [kaniko](#using-kaniko-as-an-image-builder)).
+  This can be tricky as you have to either make those images available to the k8s docker daemon or pull-able from a reachable registry, where they should be preloaded. Use `Values.registy.defaultBaseRegistryURL` to point nuclio at searching in your registry for those images, rather then at the default location of `quay.io/nuclio`.
+  To save some work on setting up a registry and preloading the onbuild images to it (or as a reference to what it should include) - take a look at the [prebaked-registry](https://github.com/nuclio/prebaked-registry).
+- For the Nuclio templates library to be available to you, you'll have to package that yourself, and have it served locally, somewhere within reach of your system. To point Nuclio to it, set `Values.dashboard.templatesArchiveAddress` to where you serve the templates.
+
+
+## Using Kaniko as an image builder
+
+When dealing with production deployments, bind-mounting the docker socket into Nuclio dashboard's pod is a bit of a no-no. Having access to the host machine's docker daemon by the Nuclio dashboard is akin to giving it root access to the machine.
+This is understandably a concern for real production use cases. Ideally, no pod should access the docker daemon directly, but since Nuclio is a docker based serverless framework, it needs the ability to build [OCI images](https://github.com/opencontainers/image-spec) at run time.
+While there are several alternatives to bind-mounting the docker socket, we have opted to integrate [Kaniko](https://github.com/GoogleContainerTools/kaniko) as a production-ready alternative to build OCI images in a secured way in latest versions.
+It is well maintained, stable, easy to use, and provides an ample set of features.
+Kaniko is available to use as of version 1.3.15 of Nuclio, currently only on k8s.
```

Comment:
You mention a version and below/above write a different one. Prefer not to mention the version number here?

Reply:
Good catch, changed explicit version in example snippets to `<version>`. Note that this refers to the still unreleased 1.3.15, where we finish ironing out the kinks of kaniko usage